src/components/shared/modals/BannyAttachModal.tsx

```diff
@@ -0,0 +1,77 @@
+import { t } from '@lingui/macro'
+import { Col, Modal, Row, Space } from 'antd'
+import { ThemeContext } from 'contexts/themeContext'
+import { useContext, useState } from 'react'
+
+type AttachableBanny = {
+  url: string
+  alt: string
+}
+
+const ATTACHABLE_BANNYS: AttachableBanny[] = [
+  { url: '/assets/banny_eth.png', alt: 'Blockchain Banny' },
+  { url: '/assets/banny_love.png', alt: 'Banny in love' },
+  { url: '/assets/banny_party.png', alt: 'Party banny' },
+  { url: '/assets/banny_popcorn.png', alt: 'Popcorn banny' },
+  { url: '/assets/quint.gif', alt: 'Quint' },
+  { url: '/assets/stoned_banny.png', alt: 'Stoned banny' },
+]
+
+export default function BannyAttachModal({
+  visible,
+  onBannySelected,
+  onClose,
+}: {
+  visible: boolean
+  onBannySelected: (url: string) => void
+  onClose: VoidFunction
+}) {
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+  const [selectedBannyIndex, setSelectedBannyIndex] = useState<number>()
+  const renderAttachableBanny = (banny: AttachableBanny, index: number) => {
```

Comment:
**suggestion**: could we make this a component that takes props instead?

```suggestion
  const AttachableBanny = ({ banny, index }): { banny: AttachableBanny, index: number }) => {
```

Reply:
Also its generally an antipattern to pass in an index to a child component - this creates an unnecessary coupling and is harder to maintain.

In this case we might:
- accept a prop `isSelected: boolean`, when true, set the backgroun color
- set the prop in the parent component (`isSelected={selectedBannyIndex === index}`)

This way the child component `AttachableBanny` is uncoupled from the parent component.

Reply:
Don't think this is relevant anymore? Or is it

Reply:
@johnnyd-eth the `index` prop was still there so I removed it in https://github.com/jbx-protocol/juice-interface/pull/1218/commits/365bcabc754a1110f40ebd626fdd8ebd4f34fc26

One tip is to never (unless absolutely necessary) apply styling via JS for perf reasons. I've added a CSS class for the hover state in https://github.com/jbx-protocol/juice-interface/pull/1218/commits/365bcabc754a1110f40ebd626fdd8ebd4f34fc26